packages/flutter/test/material/scaffold_test.dart

```diff
@@ -389,6 +389,49 @@ void main() {
     expect(scrollable.position.pixels, equals(0.0));
   }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS,  TargetPlatform.macOS }));
 
+  testWidgets('Tapping the status bar scrolls to top in nested navigator', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(platform: debugDefaultTargetPlatformOverride),
+        home: MediaQuery(
+          data: const MediaQueryData(padding: EdgeInsets.only(top: 25.0)), // status bar
+          child: Scaffold(
+            appBar: AppBar(),
+            body: Navigator(
+              onGenerateRoute: (RouteSettings settings) =>
+                  MaterialPageRoute<dynamic>(
+                builder: (BuildContext ctx) {
+                  return Scaffold(
+                    body: ListView.separated(
+                        controller:
+                            PrimaryScrollController.of(ctx, rootState: true),
```

Comment:
This does not seem like a better experience. It does not seem any different than just passing the PrimaryScrollController from the desired route.

Reply:
Yes, it's basically same with passing PrimaryScrollController, but if there are so many widgets to Scrollable widget, passing everything is very cumbersome. I think it would be better experience like we can access root navigator anywhere.

But, I knew it's wrong that PrimaryScrollController depends on material. So please let me know should I proceed or not. Then I'll find better solution. Thanks! :)

Reply:
@Piinks @SuhwanCha I agree, this concept of 'rootState` is a much better experience as opposed to having to pass PrimaryScrollController between potentially deeply nested trees and would be a welcome change.

However, in the end the `root` (pun intended) problem remains that we cannot assign this to multiple items correct?  Its still a major problem that we lose scroll to top functionality in apps with multiple tabs when they have their own tree.

Reply:
>  I knew it's wrong that PrimaryScrollController depends on material. So please let me know should I proceed or not. 

As this is, it cannot be merged, so this solution will not be able to proceed.

> Then I'll find better solution.

You are welcome to! Thanks for contributing! I do not know what the correct solution is, but this unfortunately cannot be accepted.